Userland/Libraries/LibJS/Runtime/ProxyObject.cpp

```diff
@@ -798,13 +798,24 @@ MarkedValueList ProxyObject::internal_own_property_keys() const
         return MarkedValueList { heap() };
 
     // 8. Let trapResult be ? CreateListFromArrayLike(trapResultArray, « String, Symbol »).
-    auto trap_result = create_list_from_array_like(global_object, trap_result_array, [](auto value) -> Result<void, ErrorType> {
-        if (!value.is_string() && !value.is_symbol())
-            return ErrorType::ProxyOwnPropertyKeysNotStringOrSymbol;
-        return {};
+    HashTable<StringOrSymbol> unique_keys;
+    auto trap_result = create_list_from_array_like(global_object, trap_result_array, [&global_object, &unique_keys](auto value) {
+        auto& vm = global_object.vm();
```

Comment:
```suggestion
    auto trap_result = create_list_from_array_like(global_object, trap_result_array, [&](auto value) {
```

Reply:
i did that because i vageuly remember hearing andreas telling horror stories about `[&]` but i honestly cant remember the reasons why, will fix.

Reply:
that was about capturing a stack-allocated object by reference in a lambda that is stored for later use (think callbacks in LibGUI etc.) - this is fine :)